src/Telemetry/AzureDevOpsTimeline/KustoTimelineTelemetryRepository.cs

```diff
@@ -0,0 +1,187 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Kusto.Data.Common;
+using Kusto.Data.Exceptions;
+using Kusto.Data.Net.Client;
+using Kusto.Ingest;
+using Microsoft.DotNet.Internal.AzureDevOps;
+using Microsoft.DotNet.Kusto;
+using Microsoft.DotNet.Services.Utility;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.AzureDevOpsTimeline
+{
+
+    public sealed class KustoTimelineTelemetryRepository : ITimelineTelemetryRepository
+    {
+        private readonly ILogger<KustoTimelineTelemetryRepository> _logger;
+        private readonly IKustoIngestClient _ingest;
+        private readonly ICslQueryProvider _query;
+        private readonly string _database;
+
+        public KustoTimelineTelemetryRepository(ILogger<KustoTimelineTelemetryRepository> logger, IOptionsSnapshot<KustoTimelineTelemetryOptions> options)
+        {
+            _logger = logger;
+
+            // removing the IngestConnectionString was a default setup in local debugging
+            if (string.IsNullOrEmpty(options.Value.IngestConnectionString))
+            {
+                _logger.LogDebug("No ingest connection string provided; will ignore ingest operations");
+                _ingest = new NullKustoIngestClient();
```

Comment:
Hrm... I wonder if this should be DI's correctly (either DI a Null one or a real one, and let this guy not worry about it).

That might be a pain to do though. :-) So not a big deal.